ogr/ogrsf_frmts/gpkg/ogr_geopackage.h

```diff
@@ -411,6 +414,44 @@ class GDALGeoPackageDataset final : public OGRSQLiteBaseDataSource,
 
     GDALDataset *GetRasterLayerDataset(const char *pszLayerName);
 
+    //! Instance of that class temporarily disable foreign key checks
+    class TemporaryForeignKeyCheckDisabler
+    {
+      public:
+        explicit TemporaryForeignKeyCheckDisabler(GDALGeoPackageDataset *poDS)
+            : m_poDS(poDS),
+              m_nPragmaForeignKeysOldValue(SQLGetInteger(
+                  m_poDS->GetDB(), "PRAGMA foreign_keys", nullptr))
+        {
+            if (m_nPragmaForeignKeysOldValue)
+            {
+                CPL_IGNORE_RET_VAL(
+                    SQLCommand(m_poDS->GetDB(), "PRAGMA foreign_keys = 0"));
+            }
+        }
+
+        ~TemporaryForeignKeyCheckDisabler()
+        {
+            if (m_nPragmaForeignKeysOldValue)
+            {
+                CPL_IGNORE_RET_VAL(
+                    SQLCommand(m_poDS->GetDB(), "PRAGMA foreign_keys = 1"));
```

Comment:
I think there is no need for setting that value back as that is only a connection wide setting. This connection won't be reused outside the geopackage writer, right?

Reply:
Maybe it is not expensive to turn it on, and wouldn't it be better to play safe?

Reply:
> This connection won't be reused outside the geopackage writer, right?

Users of the API could potentially run the GDALDataset::ExecuteSQL("PRAGMA foreign_keys = 1") , so better restore to the setting they used